include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,537 @@
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    const_reference at(size_type const i) const
+    {
+        if (i >= size()) // [[unlikely]]
+            throw std::out_of_range{"Trying to access element behind the last in gap_decorator."};
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    constexpr reference operator[](size_type const i) const
+    {
+        assert(sequence); // sequence must not be a nullptr
+        assert(i < size());
+
+        // case 1: there are no gaps
+        if (!anchors.size()) // [[unlikely]]
+            return value_type((*sequence)[i]);
+
+        // case 2: there are gaps
+        set_iterator_type it = anchors.upper_bound(gap_t{i, 0/*Unused*/});
+
+        if (it == anchors.begin())
+            return value_type((*sequence)[i]); // since no gaps happen before i
+
+        size_type gap_len{(*--it).second};
+
+        if (it != anchors.begin())
+            gap_len -= (*(std::prev(it))).second;
+
+        if (i < (*it).first + gap_len)
+           return gap{};
+        else
+           return value_type((*sequence)[i - (*it).second]);
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+    /*!\brief Compare seqan3::gap_decorator_anchor_set by underlying sequence and gaps.
+     * \tparam    inner_type2  The underlying type of the decorator to compare with; must be the same as inner_type
+     *                         when ignoring constness.
+     * \param[in] rhs          The right-hand side gap decorator to compare.
+     * \returns A boolean flag indicating (in)equality of the aligned sequences.
+     *
+     * \par Complexity
+     * Worst case: \f$O(n*\log k)\f$
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee. Does not modify the aligned sequences.
+     */
+    template <typename inner_type2>
+    //!\cond
+        requires std::is_same_v<std::remove_const_t<inner_type>, std::remove_const_t<inner_type2>>
+    //!\endcond
+    constexpr bool operator==(gap_decorator_anchor_set<inner_type2> const & rhs) const noexcept
+    {
+        if (anchors.size() != rhs.anchors.size())
+            return false;
+
+        if (sequence) // not nullptr
+        {
+            if (rhs.sequence) // not nullptr
+            {
+                if (sequence->size() == rhs.sequence->size())
+                    return std::ranges::equal(*this, rhs);
+            }
+        }
+        else if (!rhs.sequence)
+        {
+            return true; // both are nullptr (and both anchors should be empty as inserting gaps into a nullptr is UB)
+        }
+
+        return false;
+    }
+
+    //!\copydoc operator==
+    template <typename inner_type2>
+    //!\cond
+        requires std::is_same_v<std::remove_const_t<inner_type>, std::remove_const_t<inner_type2>>
+    //!\endcond
+    constexpr bool operator!=(gap_decorator_anchor_set<inner_type2> const & rhs) const noexcept
+    {
+        return !(operator==(rhs));
+    }
+    //!\}
```

Comment:
implement comparison operators as friends, add lexicographical comparison. I think you can use things like std::less to make this whole thing easier.